esi_leap/objects/base.py

```diff
@@ -47,7 +47,11 @@ def _attr_as_dict(field):
             if isinstance(attr, ESILEAPObject):
                 attr = attr.as_dict()
             return attr
-
-        return dict((k, getattr(self, k))
-                    for k in self.fields
-                    if self.obj_attr_is_set(k))
+        d = {}
+        for k in self.fields:
+            if self.obj_attr_is_set(k):
+                value = getattr(self, k)
+                if isinstance(value, (datetime.date, datetime.datetime)):
+                    value = value.isoformat()
```

Comment:
I personally don't love this; what if someone wants to work with the date as a date?

Reply:
Yes, you're right. I'll remove it. I shouldn't put to_string method here. Date to string is only needed when converting into marketplace offer dictionary.